jetbrains-ultimate/src-201/software/aws/toolkits/jetbrains/services/rds/auth/IamAuth.kt

```diff
@@ -51,12 +54,23 @@ class IamAuth : DatabaseAuthProvider, CoroutineScope by ApplicationThreadPoolSco
     ): CompletionStage<ProtoConnection>? {
         LOG.info { "Intercepting db connection [$connection]" }
         return future {
-            val credentials = getCredentials(connection)
-            DatabaseCredentialsAuthProvider.applyCredentials(connection, credentials, true)
+            var result = Result.Succeeded
+            val project = connection.runConfiguration.project
+            try {
+                val credentials = getCredentials(connection)
+                DatabaseCredentialsAuthProvider.applyCredentials(connection, credentials, true)
+            } catch (e: Throwable) {
+                result = Result.Failed
+                throw e
+            } finally {
+                // TODO find a more direct way to do this
```

Comment:
consider making this an extension so TODO is in one place

Reply:
Added